packages/lib/src/interactions/utils.ts

```diff
@@ -99,3 +106,52 @@ export function clampRectangleToVis(
   const shift = newCenter.clone().sub(center);
   return [startPoint.clone().add(shift), endPoint.clone().add(shift)];
 }
+
+export function getDistinctPoints(
+  startPoint: Vector2,
+  endPoint: Vector2,
+  xDomain: Domain,
+  yDomain: Domain
+): [Vector2, Vector2] {
+  const [xMin, xMax] = xDomain;
+
+  if (Math.abs(endPoint.x - startPoint.x) < Math.abs(xMax - xMin) / 200) {
+    return [new Vector2(xMin, startPoint.y), new Vector2(xMax, endPoint.y)];
+  }
+
+  const [yMin, yMax] = yDomain;
+  if (Math.abs(endPoint.y - startPoint.y) < Math.abs(yMax - yMin) / 200) {
+    return [new Vector2(startPoint.x, yMin), new Vector2(endPoint.x, yMax)];
+  }
+
+  return [startPoint, endPoint];
+}
+
+export function getZoomTransform(
+  keepRatio?: boolean,
+  fullDimIfEmpty?: boolean
+): ZoomTransformFn {
+  if (fullDimIfEmpty && keepRatio) {
+    return (
+      startPt: Vector2,
+      endPt: Vector2,
+      xDomain: Domain,
+      yDomain: Domain
+    ) =>
+      getRatioRespectingRectangle(
+        ...getDistinctPoints(startPt, endPt, xDomain, yDomain),
+        xDomain,
+        yDomain
+      );
+  }
```

Comment:
I combine both transform if needed but this is a bit over the top. There is not much point having both at the same time. Indeed, if the user draws an empty rectangle in this case and asks for the full width, the only way to keep the ratio when having the full width is to have the full height which makes the zoom meaningless.

Reply:
Yeah I agree, I don't think it's needed.